rsconnect/actions.py

```diff
@@ -1246,401 +1139,47 @@ def deploy_by_manifest(
     :return: the ultimate URL where the deployed app may be accessed and the sequence
     of log lines.  The log lines value will be None if a log callback was provided.
     """
-    app_store = AppStore(manifest_file_name)
-    (
-        app_id,
-        deployment_name,
-        deployment_title,
-        default_title,
-        app_mode,
-        _,
-        _,
-    ) = gather_basic_deployment_info_from_manifest(connect_server, app_store, manifest_file_name, new, app_id, title)
-    bundle = make_manifest_bundle(manifest_file_name)
-    return _finalize_deploy(
-        connect_server,
-        app_store,
-        manifest_file_name,
-        app_id,
-        app_mode,
-        deployment_name,
-        deployment_title,
-        default_title,
-        bundle,
-        log_callback,
-    )
-
-
-def gather_basic_deployment_info_for_notebook(
-    connect_server: api.RSConnectServer,
-    app_store: AppStore,
-    file_name: str,
-    new: bool,
-    app_id: int,
-    title: str,
-    static: bool,
-) -> typing.Tuple[int, str, str, bool, AppMode]:
-    """
-    Helps to gather the necessary info for performing a deployment.
-
-    :param connect_server: the Connect server information.
-    :param app_store: the store for the specified file
-    :param file_name: the primary file being deployed.
-    :param new: a flag noting whether we should force a new deployment.
-    :param app_id: the ID of the app to redeploy.
-    :param title: an optional title.  If this isn't specified, a default title will
-    be generated.
-    :param static: a flag to note whether a static document should be deployed.
-    :return: the app ID, name, title information and mode for the deployment.
-    """
-    validate_file_is_notebook(file_name)
-    _validate_title(title)
-
-    if new and app_id:
-        raise RSConnectException("Specify either a new deploy or an app ID but not both.")
-
-    if static:
-        app_mode = AppModes.STATIC
+    kwargs = locals()
+    kwargs["manifest_file_name"] = manifest_file_name = validate_manifest_file(manifest_file_name)
+    app_mode = read_manifest_app_mode(manifest_file_name)
+    kwargs["title"] = title or default_title_from_manifest(manifest_file_name)
+
+    if isinstance(connect_server, api.RSConnectServer):
```

Comment:
This code block occurs repeatedly - let's factor out a helper that returns a dict we can pass to `kwargs.update`.